### PR TITLE
Bump runtime to 42

### PR DESCRIPTION
**Pull request by bbhtt**
<del>I changed all the sources to release archives rather than only git commits.</del> This is blocked because svgvi uses gxml 0.18 https://gitlab.com/gsvg/svgvi/-/blob/771ec8f44bafa1d059938a82372476004e1ef7d8/meson.build#L14 but none of the other modules have a release archive with that dep.

Hey @esodan, please review when you have time, thanks!

### CODE REVIEW

File: mx.pwmc.Svgvi.json
Comment (hfiguiere) [inline mx.pwmc.Svgvi.json]: The SDK has 0.56.1. Shouldn't that be enough and not have to build this?

File: mx.pwmc.Svgvi.json
Comment (hfiguiere) [inline mx.pwmc.Svgvi.json]: this exact version is in the SDK.

Comment (bbhtt) [reply mx.pwmc.Svgvi.json]: Thanks!

File: mx.pwmc.Svgvi.json
Comment (bbhtt) [reply mx.pwmc.Svgvi.json]: Yes, probably. I removed it for now. Thanks!

**Comment by bbhtt** [comment]
(can't bump the commit in the module due to the issue I mentioned, so adding it as a patch)

**Comment by bbhtt** [comment]
> Build [102887 failed](https://buildbot.flathub.org/#/builders/32/builds/102887)

Icon size is wrong:

`error: /srv/buildbot/worker/build-x86_64-7/build/builddir/export/share/icons/hicolor/512x512/apps/mx.pwmc.Svgvi.png is not a valid icon: Expected a square icon but got: 500x512`

Seems to fixed in master, I'll update the commit to 771ec8f44bafa1d059938a82372476004e1ef7d8

Ah no... it's about the png icon. I'll open an issue and add a patch to stop installing the png icon.

**Comment by bbhtt** [comment]
Appdata was failing on: 

```
flatpak run --env=G_DEBUG=fatal-criticals org.freedesktop.appstream-glib validate *.xml.in

mx.pwmc.Svgvi.appdata.xml.in: FAILED:
• url-not-found         : <screenshot> failed to connect: SSL handshake failed [https://www.pwmc.mx/web/image/2843]
• tag-missing           : <description> required
Validation of files failed
```

There was a patch for the screenshot already but it had to be updated since it was failing to apply in https://buildbot.flathub.org/#/builders/32/builds/102892 (as I updated the commit)

**Comment by bbhtt** [comment]
bot, build

**Comment by pwmcmx** [comment]
Hey thanks for your interest. I've been porting GSvgvi to Gtk4 so using 42 runtime will be easy. Now I've finished almost its dependencies to be updated and ported to Gtk4; hope that will make easy to port GSvgvi too.

**Comment by bbhtt** [comment]
> Hey thanks for your interest. I've been porting GSvgvi to Gtk4 so using 42 runtime will be easy. Now I've finished almost its dependencies to be updated and ported to Gtk4; hope that will make easy to port GSvgvi too.

Thanks. I think this can be merged as is if there are no issues. Or do you want to wait for the GTK4 version release? In that case, it'd be great if you can fix the gxml issue mentioned above and the appdata/icon issue.

**Comment by hfiguiere** [comment]
gnome 40 is EOL in just a few weeks, so...

File: appdata-patch.patch
Comment (hfiguiere) [inline appdata-patch.patch]:
```suggestion
+  <id>mx.pwmc.Svgvi</id>
```

Don't put `desktop` in the app-id.

It should be removed upstream too.

File: remove-png-icon.patch
Comment (hfiguiere) [inline remove-png-icon.patch]: why?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: appdata-patch.patch
Comment (bbhtt) [reply appdata-patch.patch]: It was a whitespace change there I think. I'll update the patch to remove the .desktop suffix.

File: remove-png-icon.patch
Comment (bbhtt) [reply remove-png-icon.patch]: Because of wrong dimensions of the png icon https://github.com/flathub/mx.pwmc.Svgvi/pull/3#issuecomment-1200356060

Comment (hfiguiere) [reply remove-png-icon.patch]: ah ok.

you could just remove it in post-install instead.

File: appdata-patch.patch
Comment (bbhtt) [reply appdata-patch.patch]: Ok I updated the appdata with appstream-util upgrade and removed the desktop suffix.

File: remove-png-icon.patch
Comment (bbhtt) [reply remove-png-icon.patch]: Added a post install command to remove it.

**Comment by bbhtt** [comment]
I synced the changes from https://gitlab.com/gsvg/svgvi/-/commit/bfe406763c12ae252ed3961de52229a7418f85f6 This allowed to use newer gxml, release tarballs for all except one dependency and drop the vala build patch I added. 

The appdata patch should also apply cleanly now

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
> Started [test build 104981](https://buildbot.flathub.org/#/builders/32/builds/104981)

Error while building svgvi, it wants gsvgtk 0.12

```
Run-time dependency gsvgtk-0.12 found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency gsvgtk-0.12
Cloning into 'gsvgtk'...
fatal: unable to access 'https://gitlab.com/gsvg/gsvgtk.git/': Could not resolve host: gitlab.com
../meson.build:8:0: ERROR: Git command failed: ['/usr/bin/git', 'clone', 'https://gitlab.com/gsvg/gsvgtk.git', 'gsvgtk']
```

Hey, @pwmcmx I think you need to bump the version here https://gitlab.com/gsvg/gsvgtk/-/blob/dd5b7331e8b75946898b22658d9c8a4cd651a765/meson.build#L2 because gsvgtk 0.12 was built and installed properly https://buildbot.flathub.org/#/builders/35/builds/12609/steps/6/logs/stdio (17985)

**Comment by bbhtt** [comment]
bot, build

I added a patch since I received no response. The issue was pkg-config was failing on gsvg version, gsvgtk was a red herring (I think it should be updated anyways)

**Comment by esodan** [comment]
Hope I've fixed all issues to use runtime 42 and now we are using GTK 4.0 and updated dependencies. So closing this issue.

Thanks for all your help.

**Comment by bbhtt** [comment]
@esodan, It failed to build due to a timeout with the API, you should re-trigger the build.

A couple of things:

1. https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/mx.pwmc.Svgvi.json#L36 Do you need an older libgee? libgee 0.20.5 is in the runtime, if you don't then it can be removed. I used libgee 0.20.5 from the runtime after @hfiguiere suggested and it built fine.

2. I think you'll still need this patch for it to build https://github.com/flathub/mx.pwmc.Svgvi/pull/3/commits/afd7eb88cfde14b5826f495e167c51053801358b because as far as I can tell the issue is not fixed in gsvgtk

3. Please don't use `branch` 

https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/mx.pwmc.Svgvi.json#L48

https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/mx.pwmc.Svgvi.json#L59

https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/mx.pwmc.Svgvi.json#L70

Please use tag + commit if you have to or better use release tarballs if you can.

4. I removed this build flag because it wasn't valid anymore https://gitlab.com/gsvg/gsvgtk/-/blob/6b0008e1c97495730b7e8e292ae7c653f994ee57/meson_options.txt

https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/mx.pwmc.Svgvi.json#L78

It's still there in your commit.

5.  The icon size wasn't fixed https://gitlab.com/gsvg/svgvi/-/blob/03bc348fba47fa1a1ae67d532a981f21e115eb5f/src/mx.pwmc.Svgvi.png it needs to be exactly square otherwise it'll fail to build in flatpak. So you either need to cleanup the icon like I did https://github.com/flathub/mx.pwmc.Svgvi/pull/3/files#diff-6e65119598a38c3de2084398df09ecda284ebf83cf963db211fe3d6cd53a6facR114 /add a patch /update the size and commit for svgvi.

6. You should remove this file from here, it's not needed anymore https://github.com/flathub/mx.pwmc.Svgvi/blob/21dcce4c75ae0ac80cb2577610b5f0b0c461dbfe/appdata-screenshot.patch

Also please don't commit directly to master without testing if it builds properly. Otherwise it'll produce broken builds/it won't get published and there's no notification for this, so people may not notice.

**Pull request closed without merging**
